lib/rules/method.js

```diff
@@ -63,6 +63,12 @@ function checkCallExpression(ruleHelper, callExpr, node) {
         }
         break;
 
+    case "TSNonNullExpression":
```

Comment:
creating a custom/mock call expression seems to make sense here, but can you provide an example or comment that explains what exactly we're looking at?

Reply:
Could you possibly provide some more explanation on what you mean by custom/mock call expression?

As for context take a look here:
https://ts-ast-viewer.com/#code/B4QgFAlAdAlgdnApgJwBIBUCyAZABAXlwCMBDAEwxyA

Here's some background on the null coalescing operator

Taken from typescriptlang.org:

---------

Non-null assertion operator
A new ! post-fix expression operator may be used to assert that its operand is non-null and non-undefined in contexts where the type checker is unable to conclude that fact. Specifically, the operation x! produces a value of the type of x with null and undefined excluded. Similar to type assertions of the forms <T>x and x as T, the ! non-null assertion operator is simply removed in the emitted JavaScript code.

```
// Compiled with --strictNullChecks
function validateEntity(e?: Entity) {
    // Throw exception if e is null or invalid entity
}

function processEntity(e?: Entity) {
    validateEntity(e);
    let s = e!.name;  // Assert that e is non-null and access name
}
```
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#:~:text=non%2Dnull%20assertion%20expression%20operator,checking%20mode%20for%20backwards%20compatibility.

Reply:
https://github.com/mozilla/eslint-plugin-no-unsanitized/blob/370c88b589e997de45774a44766778ba76af8848/lib/rules/method.js#L68-L71 is an example of where we create a mock call expression and check it. I could envision us doing something similar for NonNull operator.
I.e. if someone calls `foo!()`, we can just pretend it's like a call to `foo()`

Reply:
looking at this now!

Reply:
Ohhh I misunderstood your original question - so without this the test case:

```
node!().insertAdjacentHTML('beforebegin', evil);
```

will not report any errors.  Node.right in this case refers to the method or property name.